tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -640,6 +649,55 @@ EOF
 		chmod g+rx,o+x "${ROOTFS_DIR}"
 	fi
 
+	if [ "${AGENT_POLICY}" == "yes" ]; then
+		# Setup systemd-based environment for kata-opa.
+		local opa_bin_dir="$(get_opa_bin_dir "${ROOTFS_DIR}")"
+		if [ -z "${opa_bin_dir}" ]; then
+			# OPA was not installed already, so download it here.
+			#
+			# TODO: if an OPA package is not available for the Guest image distro,
+			#   	Kata should cache the OPA source code, toolchain information, etc.
+			#   	OPA should be built from the cached source code instead of downloading
+			#   	this binary.
+			#
+			opa_bin_url="$(get_package_version_from_kata_yaml externals.open-policy-agent.meta.binary)"
```

Comment:
I left a few comments about this part in the versions.yaml changes.

Reply:
Thanks! New issue to track this work: https://github.com/kata-containers/kata-containers/issues/7616